controllers/nutanixcluster_controller.go

```diff
@@ -178,12 +182,21 @@ func (r *NutanixClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 	conditions.MarkTrue(cluster, infrav1.CredentialRefSecretOwnerSetCondition)
 
-	v3Client, err := CreateNutanixClient(ctx, r.SecretInformer, r.ConfigMapInformer, cluster)
+	v3Client, err := prismclientcache.DefaultCache.Get(cluster.Name)
```

Comment:
> @sid thanks for fixing this. one question, would it make sense to have per PC instead of per workload cluster client cache? sure we will have to implement ref count so that pc does not get removed if used by cluster. also understand that unique key for pc would be an issue if fqdn vs ip was provided by 2 separate clusters but that can may be fixed with get ip if fqdn is and store that? this will reduce the connection count to less as pc count will be less as compared to cluster count

No it doesn't make sense to have per PC as the users can be different per workload cluster even if the PC is the same. Overoptimization is the root of all evil; Let's stay away from that.

Reply:
sure username ca be added as a part of connection too. only concern is from IAM side whether it can handle say 200 clusters having one connection per cluster. we can take it in separate PR if needed